### PR TITLE
Open source pkg/config

### PR DESCRIPTION
**Pull request by instabledesign**
- [x] Write README

TODO in another PR
- Add Dependabot
- Add github actions
- Add Codecov
- Add Linter

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+test:
+	@echo "> Launch tests ..."
+	CGO_ENABLED=0 go test -count=1 -v ./...
```
Comment (eko) [inline Makefile]: Maybe we could add `-race` option?

**Review by eko — DISMISSED** [review]

**Review by qneyrat — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -1 +1,98 @@
-# go-config
\ No newline at end of file
+# e-TF1 go-config
+
+Package e-TF1 go-config allow you to load your application config from multiple source (dotenv, env, flags...).
```
Comment (eko) [inline README.md]: `allow` -> `allows`
`config` -> `configuration`
`source` -> `sources`

File: README.md [hunk 3]
```diff
@@ -1 +1,98 @@
-# go-config
\ No newline at end of file
+# e-TF1 go-config
+
+Package e-TF1 go-config allow you to load your application config from multiple source (dotenv, env, flags...).  
+This package is build on top of [confita](https://github.com/heetch/confita)
```
Comment (eko) [inline README.md]: `build` -> `built`

File: README.md [hunk 4]
```diff
@@ -1 +1,98 @@
...
+-----------------------------------
+       Debug|                false|bool `config:"DEBUG"`
+    HTTPAddr|                :8001|string `config:"HTTP_ADDR"`
+ BDDPassword| *** Hidden value ***|string `config:"BDD_PASSWORD" print:"-"`
+-----------------------------------
+```
+
+# Loaders 
+
+The library provider provide a DefaultConfigLoader that load from
```
Comment (eko) [inline README.md]: `provide` -> `provides`

File: README.md [hunk 5]
```diff
@@ -1 +1,98 @@
...
+// and even prepend multiple backends to it
+f := ".env"
+if _, err := os.Stat(f); err == nil {
+    cl.PrependBackends(dotenv.NewBackend(f))
+}
+```
+
+# Printer
+
+The library provide you a config table printer. A special tag `print:"-"` prevent config value to be printed.
```
Comment (eko) [inline README.md]: `provide` -> `provides`
`prevent` -> `prevents`

**Review by eko — DISMISSED** [review]

File: README.md [hunk 6]
```diff
@@ -1 +1,98 @@
...
+type MyConfig struct {
+	Debug    bool   `config:"DEBUG"`
+	HTTPAddr string `config:"HTTP_ADDR"`
+	BDDPassword string `config:"BDD_PASSWORD" print:"-"` // with print:"-" it will be print as "*** Hidden value ***"
+}
+
+func New(context context.Context) *MyConfig {
+	cfg := &MyConfig{
+		HTTPAddr: ":8001",
+        BDDPassword: "my_fake_password",
```
Comment (pimousse1099) [inline README.md]: a little tab is missing here 👍

File: README.md [hunk 7]
```diff
@@ -1 +1,98 @@
...
+-----------------------------------
+       Debug|                false|bool `config:"DEBUG"`
+    HTTPAddr|                :8001|string `config:"HTTP_ADDR"`
+ BDDPassword| *** Hidden value ***|string `config:"BDD_PASSWORD" print:"-"`
+-----------------------------------
+```
+
+# Loaders 
+
+The library provides a DefaultConfigLoader that load from
```
Comment (pimousse1099) [inline README.md]: `load` -> `loads`

**Review by pimousse1099 — CHANGES_REQUESTED** [review]
I might be wrong but I think there is another **big difference** with `confita` loading sytem.

let's say we have a `.env` file :
```
MY_KEY_1="value1"
MY_KEY_2="value2"
```

and then you run your app like this:
```
go run main -my-key-1 value-from-flags
```

**With CONFITA, the config will only load the last backend values**, so it will look like this:

```
MY_KEY_1="value-from-flags"
```

whereas with this library, the config will look like this:
```
MY_KEY_1="value-from-flags"
MY_KEY_2="value2"
```

I think we must explain this in the README

File: README.md [hunk 8]
```diff
@@ -1 +1,98 @@
-# go-config
\ No newline at end of file
+# e-TF1 go-config
+
+Package e-TF1 go-config allows you to load your application configuration from multiple sources (dotenv, env, flags...).  
+This package is built on top of [confita](https://github.com/heetch/confita) 
+
+The main difference with confita is that the last loader that matches the config key will override the previous match.
```
Comment (pimousse1099) [inline README.md]: maybe add something like this

```suggestion
The main differences with confita are the following:
* the final config will aggregate all the loaders results.
* the last loader that matches the config key will override the previous match.
```

Comment (pimousse1099) [reply README.md]: but I'm still not sure that a phrase is clear enough compared to a demo or some code 👍

File: README.md [hunk 9]
```diff
@@ -3,7 +3,11 @@
 Package e-TF1 go-config allows you to load your application configuration from multiple sources (dotenv, env, flags...).  
 This package is built on top of [confita](https://github.com/heetch/confita) 
 
-The main difference with confita is that the last loader that matches the config key will override the previous match.
+The main differences with confita are the following:
+
+ - all the backend matched value are write on given struct
```
Comment (pimousse1099) [inline README.md]:
```suggestion
 - all the backend matched values are written on a given struct 
```

isn't it already the case for confita ?

File: README.md [hunk 10]
```diff
@@ -3,7 +3,11 @@
 Package e-TF1 go-config allows you to load your application configuration from multiple sources (dotenv, env, flags...).  
 This package is built on top of [confita](https://github.com/heetch/confita) 
 
-The main difference with confita is that the last loader that matches the config key will override the previous match.
+The main differences with confita are the following:
+
+ - all the backend matched value are write on given struct 
+ - the last loader that matches the config key will override the previous match
+ - we allow to override by empty string
```
Comment (pimousse1099) [inline README.md:10]: 👍

**Review by pimousse1099 — CHANGES_REQUESTED** [review]
the last commit does not cover this case : https://github.com/etf1/go-config/pull/1#pullrequestreview-375975096

yet the aditon for the empty value support is great !

**Review by pimousse1099 — APPROVED** [review]

**Review by eko — APPROVED** [review]

**Review by qneyrat — APPROVED** [review]

**Pull request merged**